qsc/plot.py

```diff
@@ -0,0 +1,529 @@
+        # Define the rotation arrays for the subplots
+        degrees_array_x = [0., -66., 81.] # degrees for rotation in x
+        degrees_array_z = [azim_default, azim_default, azim_default] # degrees for rotation in z
+        shift_array   = [-1.0, 0.7, 1.8]
+        # Import mayavi and rotation packages (takes a few seconds)
+        from mayavi import mlab
+        from scipy.spatial.transform import Rotation as R
+        # Show RZ plot
+        plt.show()
+        # Close RZ plot
+        plt.close()
+        # Create 3D figure
+        fig = mlab.figure(bgcolor=(1,1,1), size=(430,720))
+        # Create subplots
+        create_subplot_mayavi(mlab, R, alphas, x_2D_plot, y_2D_plot, z_2D_plot,
+                              fieldline_X, fieldline_Y, fieldline_Z,
+                              Bmag, degrees_array_x, degrees_array_z, shift_array)
+        # Create a good camera angle
+        mlab.view(azimuth=0, elevation=0, distance=8.5, focalpoint=(-0.15,0,0), figure=fig)
+        # Create the colorbar and change its properties
+        cb = mlab.colorbar(orientation='vertical', title='|B| [T]', nb_labels=7)
+        cb.scalar_bar.unconstrained_font_size = True
+        cb.label_text_property.font_family = 'times'
+        cb.label_text_property.bold = 0
+        cb.label_text_property.font_size=24
+        cb.label_text_property.color=(0,0,0)
+        cb.title_text_property.font_family = 'times'
+        cb.title_text_property.font_size=34
+        cb.title_text_property.color=(0,0,0)
+        cb.title_text_property.bold = 1
+        # Save figure
+        if savefig != None:
+            mlab.savefig(filename=savefig+'3D_fieldlines.png', figure=fig)
+        # Show mayavi plot
+        mlab.show()
+        # Close mayavi plots
+        mlab.close(all=True)
+
+def B_fieldline(self, r=0.1, alpha=0, phimax=None, nphi=400):
+    '''
+    Plot the modulus of the magnetic field B along a field line with
+    the Boozer toroidal angle varphi acting as a field-line following
+    coordinate
+
+    Args:
+      r (float): near-axis radius r where to create the surface
+      alpha (float): Field-line label
+      phimax (float): Maximum value of the field-line following parameter varphi
+      nphi (int): resolution of the phi grid
+    '''
+    if phimax is None:
+        phimax = 10 * np.pi / abs(self.iota)
+    varphi_array = np.linspace(0, phimax, nphi)
+    _, ax = plt.subplots(1, 1, figsize=(10, 6), dpi=80, facecolor='w', edgecolor='k')
+    plt.xlabel(r'$\varphi$')
+    plt.ylabel(r'$B(\varphi)$')
+    plt.title("r = " + str(r) + ", alpha = " + str(alpha))
+    theta = alpha + self.iota * varphi_array
+    plt.plot(varphi_array, self.B_mag(r, theta, varphi_array, Boozer_toroidal=True))
+    #ax.xaxis.set_major_formatter(tck.FormatStrFormatter('%g $\pi$'))
+    #ax.xaxis.set_major_locator(tck.MultipleLocator(base=phimax*abs(self.iota)/np.pi))
```

Comment:
Here, and in `B_contour`, I removed these pi's that were printed on the axes because it made the labels incorrect. They were just putting pi after the true value, e.g. the axis had a tick label "phi=5pi" where the tick was really at phi=5. Feel free to reinstate the pi's if you fix this.

Reply:
I fixed this by dividing the x array by pi.